### PR TITLE
[WIP] Extend analysis to include fully-interacting system

### PR DESCRIPTION
**Pull request by jchodera**
This includes a bugfix for crashes that would occur while writing to the NetCDF file in MPI mode.

### CODE REVIEW

File: examples/hydration-smiles/freesolv.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ molecules:
 
 solvents:
   water:
-    nonbonded_method: PME
+    nonbonded_method: CutoffPeriodic
```
Comment (jchodera) [inline examples/hydration-smiles/freesolv.yaml]: I'm just debugging this at the moment. Will change back to PME later.

File: Yank/sampling.py [hunk 2]
```diff
@@ -789,7 +789,10 @@ def _initialize_netcdf(self):
 
     def _write_iteration_netcdf(self):
         super(ModifiedHamiltonianExchange, self)._write_iteration_netcdf()
-        if self.fully_interacting_state is not None:
+        if self.mpicomm:
+            # Only the root node will write data.
+            if self.mpicomm.rank != 0: return
```
Comment (andrrizzi) [inline Yank/sampling.py]: I've already added this in PR #442 with another couple of bugfixes. I wanted to implement also the detection of `exclusive` vs `shared` mode before merging, but I can do it right now and you can debug from that one so that we'll avoid having to solve conflicts.

Comment (jchodera) [inline Yank/sampling.py]: That would be great! Thanks!

Comment (jchodera) [inline Yank/sampling.py]: Alternatively, autodetection could certainly go into a separate PR.

**Comment by jchodera** [comment]
Closing this to open a new PR.

File: Yank/sampling.py (hunk 2 above)
Comment (andrrizzi) [inline Yank/sampling.py]: > Alternatively, autodetection could certainly go into a separate PR.

Oh! Sorry, that's what I meant. Now that I read what I wrote above, it's a mess. Anyway now I've merged my PR so you can fetch from upstream and start from there.

Maybe it would be useful to use branches instead of forks internally, so that anyone can modify the PR of somebody else.

Comment (jchodera) [inline Yank/sampling.py]: Agreed! We should probably protect master and switch to using feature branches instead

Comment (andrrizzi) [inline Yank/sampling.py]: > We should probably protect master

Done!
